### ISSUE TITLE
Training time and GPU information

### ISSUE THREAD

**Issue by yswang1717** [weiguoPian/AV-CIL_ICCV2023#2]
Could you please share the required time for training in each experiment and relevant GPU information? Thanks !

**Comment by yswang1717** [comment]
It would be really appreciated if you also share the required GPU memory to reproduce the paper performance.

**Comment by weiguoPian** [comment]
Hello,
Thank you for your interest in our work.
We run all the experiments on a single RTX A5000 GPU with 24GB GPU memory.
With the pre-extracted features and an extire A5000 GPU, it only takes several hours to run one single experiment.

Bests,

**Comment by yswang1717** [comment]
Thanks for the quick response!

**Comment by Redredhorse** [comment]
Hello, have you successfully reached the author's experimental data? My accuracy is only 52%

**Comment by Redredhorse** [comment]
Hello, have you successfully reached the author's experimental results? My accuracy is only 52%

**Comment by weiguoPian** [comment]
> Hello, have you successfully reached the author's experimental results? My accuracy is only 52%

Hello,

Do you run the experiments with our default setting?

Bests,

**Comment by Redredhorse** [comment]
> > 您好，请问您成功达到作者的实验结果了吗？我的准确率只有52%
> 
> 你好，
> 
> 您是否使用我们的默认设置运行实验？
> 
> 最好的，

![屏幕截图_20230204_110115](https://github.com/weiguoPian/AV-CIL_ICCV2023/assets/120243984/13f84c54-e88f-4038-b805-9fda88d76a83)
I ran train_incremental_ours.py. As shown above, here are the results of my experiment. The GPU I use is V100. I refer to the settings in the paper. My experimental setup is shown below. I am very confused why the experimental result is only 52%. Hope to get your answer.
![image](https://github.com/weiguoPian/AV-CIL_ICCV2023/assets/120243984/5f996d6f-746b-4c34-89ff-eb7453e46e79)

**Comment by Redredhorse** [comment]
> > 您好，请问您成功达到作者的实验结果了吗？我的准确率只有52%
> 
> 你好，
> 
> 您是否使用我们的默认设置运行实验？
> 
> 最好的，
![image](https://github.com/weiguoPian/AV-CIL_ICCV2023/assets/120243984/41fc11e4-2983-44dc-9ed5-a00c7bbbcbb4)

I ran train_incremental_ours.py. As shown above, here are the results of my experiment. The GPU I use is V100. Regarding audio and video feature extraction, I directly use the extracted features you shared.I refer to the settings in the paper. My experimental settings is shown below. I am very confused why the experimental result is only 52%. Hope to get your answer.Thank you so much.
![image](https://github.com/weiguoPian/AV-CIL_ICCV2023/assets/120243984/4cf29dfa-5571-4f2d-a11d-a0cdd8b4f995)

**Comment by weiguoPian** [comment]
> > > 您好，请问您成功达到作者的实验结果了吗？我的准确率只有52%
> > 
> > 
> > 你好，
> > 您是否使用我们的默认设置运行实验？
> > 最好的，
> > ![image](https://private-user-images.githubusercontent.com/120243984/302018030-41fc11e4-2983-44dc-9ed5-a00c7bbbcbb4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY5NDIwNDksIm5iZiI6MTcwNjk0MTc0OSwicGF0aCI6Ii8xMjAyNDM5ODQvMzAyMDE4MDMwLTQxZmMxMWU0LTI5ODMtNDRkYy05ZWQ1LWEwMGM3YmJiY2JiNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIwM1QwNjI5MDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04MTc5Y2E2MDY0MTNhNzViMTBjNzY3MDIwMzA3NjAwOWYzYTk1NzlmMDk5MzQ1OWQwODMxODQ4MDg2NjA4ZmNkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.0iXwSmvk4C8XXAQwTAlOJP6tRxR2PkocyHoD-IebUCg)
> 
> I ran train_incremental_ours.py. As shown above, here are the results of my experiment. The GPU I use is V100. Regarding audio and video feature extraction, I directly use the extracted features you shared.I refer to the settings in the paper. My experimental settings is shown below. I am very confused why the experimental result is only 52%. Hope to get your answer.Thank you so much. ![image](https://private-user-images.githubusercontent.com/120243984/302018147-4cf29dfa-5571-4f2d-a11d-a0cdd8b4f995.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY5NDIwNDksIm5iZiI6MTcwNjk0MTc0OSwicGF0aCI6Ii8xMjAyNDM5ODQvMzAyMDE4MTQ3LTRjZjI5ZGZhLTU1NzEtNGYyZC1hMTFkLWEwY2RkOGI0Zjk5NS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIwM1QwNjI5MDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02ZjlmNWFiYmRkMmIxYTMxYzhjNGFmMzEyYjgwYWI3YmQ4MzhmM2ZmZDM3ZGE1YzZlY2NhNTk1OTE3MzA0NWNmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.KidHX4uK69tWTOPFQnWHw8RtP81oHISFuVGlCAkzRE4)

Thank you for your reply.

It seems that the hyperparameters of 'instance_contrastive', 'class_contrastive',  and 'attn_score_distil' are all set to False in your experiment.

You can use our provided [run_incremental_ours.sh](https://github.com/weiguoPian/AV-CIL_ICCV2023/blob/main/run_incremental_ours.sh) to run your experiment directly. Please refer to [sh run_incremental_ours.sh 'dataset'](https://github.com/weiguoPian/AV-CIL_ICCV2023?tab=readme-ov-file#training--evaluation).

Bests,

**Comment by adityath1** [comment]
I used your provided [run_incremental_ours.sh](https://github.com/weiguoPian/AV-CIL_ICCV2023/blob/main/run_incremental_ours.sh) to run the experiment. The results are quite different. I am quite confused.
![Screenshot from 2025-01-13 15-52-54](https://github.com/user-attachments/assets/78f0f895-5b72-42d9-a7ed-c1441b2af28d)
The GPU I am using is NVIDIA A100. Also, these results are for batch sizes: train=64, infer=32.
After increasing the batch size, the accuracy remains almost the same but avg forgetting decreases.
Hope to get your answer asap.
Thanks

**Comment by weiguoPian** [comment]
> I used your provided [run_incremental_ours.sh](https://github.com/weiguoPian/AV-CIL_ICCV2023/blob/main/run_incremental_ours.sh) to run the experiment. The results are quite different. I am quite confused. ![Screenshot from 2025-01-13 15-52-54](https://private-user-images.githubusercontent.com/185410529/402503594-78f0f895-5b72-42d9-a7ed-c1441b2af28d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY3NjUyNDQsIm5iZiI6MTczNjc2NDk0NCwicGF0aCI6Ii8xODU0MTA1MjkvNDAyNTAzNTk0LTc4ZjBmODk1LTViNzItNDJkOS1hN2VkLWMxNDQxYjJhZjI4ZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDExM1QxMDQyMjRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lYzkxNDVmODI4NjBhMWJlNGY5ZGJiNjdhMDZlZGNiODkxMjU3OGE5NTIwODE1OGEyNGVlYWExOTMzYTlkNTQ5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.0ICM8QGgeDPOzytgo-t79y55E0yYmCC5XvYdkCwn8po) The GPU I am using is NVIDIA A100. Also, these results are for batch sizes: train=64, infer=32. After increasing the batch size, the accuracy remains almost the same but avg forgetting decreases. Hope to get your answer asap. Thanks

Hello,

We set the both the exemplar_batch_size and train_batch_size to 128 in our experiments. Additionally, since the performance is constrained by the upper bound, you can also firstly evaluate the upper bound performance in your experiments environment.

Bests,

**Comment by adityath1** [comment]
for  upper bound, I have just changed class_num_per_step to 28. With this I am getting test result as 70.8 and it is not proceeding further as all the classes have been given at once. Is this the way you have calculated the upper bound? If not, Could you please provide the training code for the upper bound?
Thanks
![image](https://github.com/user-attachments/assets/c9039f77-ef3b-4e1f-ba67-042c66bcd8a7)

**Comment by weiguoPian** [comment]
> for  upper bound, I have just changed class_num_per_step to 28. With this I am getting test result as 70.8 and it is not proceeding further as all the classes have been given at once. Is this the way you have calculated the upper bound? If not, Could you please provide the training code for the upper bound?
> 
> Thanks
> 
> ![image](https://github.com/user-attachments/assets/c9039f77-ef3b-4e1f-ba67-042c66bcd8a7)
> 
> 

Please refer to https://github.com/weiguoPian/AV-CIL_ICCV2023?tab=readme-ov-file#training--evaluation for the training of upper bound. The evaluation metric used for upper bound is also the mean accuracy.

**Comment by adityath1** [comment]
> > for  upper bound, I have just changed class_num_per_step to 28. With this I am getting test result as 70.8 and it is not proceeding further as all the classes have been given at once. Is this the way you have calculated the upper bound? If not, Could you please provide the training code for the upper bound?
> > Thanks
> > ![image](https://private-user-images.githubusercontent.com/185410529/402530360-c9039f77-ef3b-4e1f-ba67-042c66bcd8a7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY4MzQwMTYsIm5iZiI6MTczNjgzMzcxNiwicGF0aCI6Ii8xODU0MTA1MjkvNDAyNTMwMzYwLWM5MDM5Zjc3LWVmM2ItNGUxZi1iYTY3LTA0MmM2NmJjZDhhNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTE0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDExNFQwNTQ4MzZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00YjRhYTY1MTRhNmMwNWRkMDAzMjM5ZWYwYTk1YjM5MmUxM2ZhOGExMmIyZjg4MDVhMDQwYzQ5MjVlMGMwMzRkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.5Qh0JxanE9v5F0EeUsk7VKX4crMBAOvhXTpMFOEzUQQ)
> 
> Please refer to https://github.com/weiguoPian/AV-CIL_ICCV2023?tab=readme-ov-file#training--evaluation for the training of upper bound. The evaluation metric used for upper bound is also the mean accuracy.
I am getting the mean accuracy for upper bound as 75.34%.

**Comment by adityath1** [comment]
I am getting the mean accuracy for upper bound as 75.34%, still my results for AVE-CI dataset does not align with the upper bound.
Hope to get your answer asap.

thanks

**Comment by weiguoPian** [comment]
> I am getting the mean accuracy for upper bound as 75.34%, still my results for AVE-CI dataset does not align with the upper bound.
> 
> Hope to get your answer asap.
> 
> 
> 
> thanks
> 
> 

I think your result is fine. The variations are likely due to differences in experimental environments, such as library versions, GPU models, and similar factors.
